### PR TITLE
save bio and avatar

### PR DESCRIPTION
**Pull request by Tomqu2012**

### CODE REVIEW

File: server/routes/user.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        return {
+            filename: `${req.body.username}-avatar-${Date.now()}`,
+            bucketName: 'avatars',
+        };
+    },
+});
+const upload = multer({ storage });
+
+// Define user schema with bio field
+const userSchema = new mongoose.Schema({
```
Comment (jacksonbarr1) [inline server/routes/user.js:34]: We have a user schema in `models/user.js` with more details that you can import

**Review by jacksonbarr1 — CHANGES_REQUESTED** [review]

File: server/routes/user.js (hunk 1 above)
Comment (Tomqu2012) [reply server/routes/user.js:34]: got it

File: server/routes/user.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+router.post('/user/bio', async (req, res) => {
+    const { username, bio } = req.body;
+
+    if (!username || !bio) {
+        return res.status(400).json({ message: 'Username and bio are required.' });
+    }
+
+    try {
+        let user = await User.findOne({ username });
+        if (!user) {
```
Comment (jacksonbarr1) [inline server/routes/user.js:54]: If the user doesn't exist we probably want to return an error and not create a new user

File: server/routes/user.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        await user.save();
+        res.status(200).json({ message: 'Bio saved successfully.', user });
+    } catch (err) {
+        console.error(err);
+        res.status(500).json({ message: 'Error saving bio.' });
+    }
+});
+
+// Save avatar endpoint
+router.post('/user/avatar', upload.single('avatar'), (req, res) => {
```
Comment (jacksonbarr1) [inline server/routes/user.js:69]: Is this working and tested? We'll definitely need a `GET` avatar too to render it on the frontend

**Pull request closed without merging**